### PR TITLE
Add pre-commit for flutter/dart

### PR DESCRIPTION
**Pull request by cjuub**

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -32,11 +32,17 @@ jobs:
...
               with:
                   python-version: '3.10'
-            - name: Install pre-commit
-              run: pip install pre-commit==3.6.0
+            - name: Install CI tools
+              run: |
+                  pip install pre-commit==3.6.0
+
+                  # Install dart from flutter
+                  git clone --depth=1 https://github.com/flutter/flutter.git
```
Comment (robinlinden) [inline .github/workflows/ci.yaml:40]: Maybe clone a specific tag so upgrades are a bit more controlled?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -15,3 +15,8 @@ repos:
       rev: v17.0.6
       hooks:
           - id: clang-format
+    - repo: https://github.com/muya/dart-format-pre-commit # See https://github.com/Cretezy/dart-format-pre-commit/pull/1
+      rev: b1c57eb1447a2d25c618017576b94793fd7aa37c
```
Comment (robinlinden) [inline .pre-commit-config.yaml:19]: Bear in mind that if muya ever force-pushes that branch to e.g. update the PR, or if the PR is merged in any way that isn't adding a merge commit, that rev will go away.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (cjuub) [reply .github/workflows/ci.yaml]: yes, just moved these lines but might as well do it now. thx

File: .pre-commit-config.yaml (hunk 2 above)
Comment (cjuub) [reply .pre-commit-config.yaml:19]: yes, maintainer seem slow as hell, but was "official" repo for this hook listed on pre-commit web page :(

did not find any better alternatives. everything is shit as always with these things

**Pull request merged**